contrib/golang/common/dso/dso.h

```diff
@@ -17,54 +17,94 @@ class DsoInstance {
 public:
   DsoInstance(const std::string dso_name);
   ~DsoInstance();
+  bool loaded() { return loaded_; }
+
+protected:
+  const std::string dso_name_;
+  void* handler_{nullptr};
```

Comment:
`handler_` It can only be read-only externally, and now becomes writable in the derived class. So it is safer to provide a function for export it. such as:
```
void* handler() {
    return handler_;
}

void setLoad(bool loaded) { loaded_ = loaded; }
```

Reply:
I think we keep the same safety for `HttpFilterDsoInstance`, that no one could update `handler_` through `HttpFilterDsoInstance`'s API.

For `DsoInstance`, it's their responsibility to take care of `handle_` those who are inheriting `DsoInstance`, as well as using the private member.

So, I prefer using `protected` here.